crates/distribution-types/src/error.rs

```diff
@@ -14,8 +14,11 @@ pub enum Error {
     #[error(transparent)]
     WheelFilename(#[from] distribution_filename::WheelFilenameError),
 
-    #[error("Unable to extract filename from URL: {0}")]
-    UrlFilename(Url),
+    #[error("Unable to extract file path from URL: {0}")]
+    MissingFilePath(Url),
+
+    #[error("Could not extract path segments")]
```

Comment:
Weird that this omits the URL, but `MissingFilePath` includes it, because the clients of these errors sometimes include the URL and sometimes don't. Specifically, where we return `MissingFilePath`, we need to include the URL; but where we return `MissingPathSegments`, we want to exclude it, because we wrap it in a message that includes the URL (so including it here would duplicate it). I don't know what the right strategy is.

Reply:
Okay, changed these to _both_ include the URL, and removed the "wrap the error in `failed to extract filename from URL`" behavior from `lock.rs`.